src/main/java/com/hrznstudio/titanium/material/ResourceTypeBlock.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * This file is part of Titanium
+ * Copyright (C) 2019, Horizon Studio <contact@hrznstudio.com>.
+ *
+ * This code is licensed under GNU Lesser General Public License v3.0, the full license text can be found in LICENSE.txt
+ */
+
+package com.hrznstudio.titanium.material;
+
+import com.google.gson.JsonObject;
+import com.hrznstudio.titanium.Titanium;
+import com.hrznstudio.titanium.api.IFactory;
+import com.hrznstudio.titanium.api.material.IHasColor;
+import com.hrznstudio.titanium.api.material.IResourceHolder;
+import com.hrznstudio.titanium.api.material.IResourceType;
+import com.hrznstudio.titanium.block.BlockBase;
+import com.hrznstudio.titanium.recipe.generator.IJSONGenerator;
+import com.hrznstudio.titanium.recipe.generator.IJsonFile;
+import net.minecraft.block.Block;
+import net.minecraft.item.BlockItem;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.text.ITextComponent;
+
+import javax.annotation.Nullable;
+import java.util.Objects;
+
+public class ResourceTypeBlock extends BlockBase implements IJsonFile, IJSONGenerator, IResourceHolder, IHasColor {
+
+    private final ResourceMaterial resourceMaterial;
+    private final IResourceType resourceType;
+    private final IAdvancedResourceType blockResourceType;
+
+    public ResourceTypeBlock(ResourceMaterial material, IResourceType type, IAdvancedResourceType blockType, ResourceTypeProperties<Properties> properties) {
+        super(material.getMaterialType() + "_" + type.getName(), (properties == null ? ((ResourceTypeProperties<Properties>) ResourceTypeProperties.DEFAULTS.get(Block.class)).get() : properties.get()));
+        this.resourceMaterial = material;
+        this.resourceType = type;
+        this.blockResourceType = blockType;
+        setItemGroup(ResourceRegistry.RESOURCES);
```

Comment:
I thought ItemGroup was part of the Block Properties?

Reply:
No, it needs to be added to the ItemBlock